v2/yarpcthrift/register.go

```diff
@@ -92,28 +71,17 @@ func BuildProcedures(s Service, opts ...RegisterOption) []transport.Procedure {
 		proto = rc.Protocol
 	}
 
-	rs := make([]transport.Procedure, 0, len(s.Methods))
+	rs := make([]yarpc.Procedure, 0, len(s.Methods))
 
 	for _, method := range s.Methods {
-		var spec transport.HandlerSpec
-		switch method.HandlerSpec.Type {
-		case transport.Unary:
-			spec = transport.NewUnaryHandlerSpec(thriftUnaryHandler{
-				UnaryHandler: method.HandlerSpec.Unary,
-				Protocol:     proto,
-				Enveloping:   rc.Enveloping,
-			})
-		case transport.Oneway:
-			spec = transport.NewOnewayHandlerSpec(thriftOnewayHandler{
-				OnewayHandler: method.HandlerSpec.Oneway,
-				Protocol:      proto,
-				Enveloping:    rc.Enveloping,
-			})
-		default:
-			panic(fmt.Sprintf("Invalid handler type for %T", method))
-		}
+		var spec yarpc.HandlerSpec
```

Comment:
Let's still keep a check to make sure we're only ever registering `Unary` handlers.

Reply:
talked offline 👍